app/controllers/activitypub/shares_controller.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+class ActivityPub::SharesController < ActivityPub::BaseController
+  include Authorization
+
+  vary_by -> { 'Signature' if authorized_fetch_mode? }
+
+  before_action :require_account_signature!, if: :authorized_fetch_mode?
+  before_action :set_status
+
+  def index
+    expires_in 0, public: public_fetch_mode?
```

Comment:
We do not want to leak information about private statuses, so I think this should be changed to:
```suggestion
    expires_in 0, public: @status.distributable? && public_fetch_mode?
```